dulwich/tests/test_porcelain.py

```diff
@@ -1317,6 +1320,222 @@ def test_hard_commit(self):
         self.assertEqual([], changes)
 
 
+class UnstageTests(PorcelainTestCase):
```

Comment:
Since unstage lives on Repo, please add these tests to dulwich/tests/test_repo.py